src/incubator/ChipsInput/index.tsx

```diff
@@ -0,0 +1,224 @@
+
+  const onChipPress = useCallback(({customValue: index}) => {
+    const selectedChip = chips[index];
+    selectedChip?.onPress?.();
+
+    setMarkedForRemoval(index);
+  },
+  [chips]);
+
+  const onChangeText = useCallback(value => {
+    fieldValue.current = value;
+    props.onChangeText?.(value);
+
+    if (!isUndefined(markedForRemoval)) {
+      setMarkedForRemoval(undefined);
+    }
+  },
+  [props.onChangeText, markedForRemoval]);
+
+  const onKeyPress = useCallback((event: NativeSyntheticEvent<TextInputKeyPressEventData>) => {
+    props.onKeyPress?.(event);
+    const keyCode = event?.nativeEvent?.key;
+    const pressedBackspace = keyCode === Constants.backspaceKey;
+
+    if (pressedBackspace && !fieldValue.current) {
+      if (isUndefined(markedForRemoval) || markedForRemoval !== chips.length - 1) {
+        setMarkedForRemoval(chips.length - 1);
+      } else {
+        removeMarkedChip();
+      }
+    }
+  },
+  [chips, props.onKeyPress, markedForRemoval, removeMarkedChip]);
+
+  const chipList = useMemo(() => {
+    return (
+      <>
+        {leadingAccessory}
+        {map(chips, (chip, index) => {
+          const isMarkedForRemoval = index === markedForRemoval;
+          return (
+            <Chip
+              key={index}
+              customValue={index}
+              // resetSpacings
+              // paddingH-s2
+              marginR-s2
+              marginB-s2
+              dismissIcon={removeIcon}
+              {...defaultChipProps}
+              {...chip}
+              onPress={onChipPress}
+              onDismiss={isMarkedForRemoval ? removeMarkedChip : undefined}
+            />
+          );
+        })}
+      </>
+    );
+  }, [chips, leadingAccessory, defaultChipProps, removeMarkedChip, markedForRemoval]);
+
+  return (
+    <TextField
+      // @ts-expect-error
+      ref={fieldRef}
+      leadingAccessory={chipList}
+      blurOnSubmit={false}
+      {...others}
+      onChangeText={onChangeText}
+      onSubmitEditing={addChip}
+      fieldStyle={[fieldStyle, styles.fieldStyle]}
+      onKeyPress={onKeyPress}
+      accessibilityHint={props.editable ? 'press keyboard delete button to remove last tag' : undefined}
+
+      // text80
+      // {...others}
+      // maxLength={undefined}
+      // title={this.props.chips ? undefined : title}
+      // value={value}
+      // onKeyPress={this.onKeyPress}
+      // enableErrors={false}
+      // onFocus={this.onFocus}
+      // onBlur={this.onBlur}
+      // hideUnderline
+      // selectionColor={isLastTagMarked ? 'transparent' : selectionColor}
+      // style={[inputStyle, styles.alignTextCenter]}
+      // containerStyle={{flexGrow: 0}}
+      // collapsable={false}
+    />
+  );
+};
+
+const styles = StyleSheet.create({
+  fieldStyle: {
+    flexWrap: 'wrap'
+  }
+});
+
+ChipsInput.changeReasons = {
+  ADDED: 'added',
+  REMOVED: 'removed'
+};
+
+export default forwardRef(ChipsInput);
+
+/* Old ChipsInput props to make sure we have parity */
```

Comment:
1. Seems like we don't have parity, are we eliminating features? 
2. can we remove all those commented props?

Reply:
True, I didn't want to add full parity cause some old props are not really needed. 
I kept them here as a comment just for the migration phase just in case there is something missing I missed out.